pkg/transaction/backend.go

```diff
@@ -50,7 +50,7 @@ func IsSynced(ctx context.Context, backend Backend, maxDelay time.Duration) (boo
 	}
 
 	header, err := backend.HeaderByNumber(ctx, big.NewInt(int64(number)))
-	if err != nil {
+	if errors.Is(err, ethereum.NotFound) {
```

Comment:
this is what you want here. in your code it might panic if err is not nil and not not found.
```go
if err != nil {
  if errors.Is(err, ethereum.NotFound) {
    return false, time.Time{}, nil
  }
  return false, time.Time{}, err
}
```